pkgs/tools/graphics/directx-shader-compiler/default.nix

```diff
@@ -0,0 +1,50 @@
+{ stdenv, fetchFromGitHub, cmake, python3, git }:
+
+stdenv.mkDerivation rec {
+  pname = "directx-shader-compiler";
+  version = "1.5.2010";
+
+  # Put headers in dev, there are lot of them which aren't necessary for
+  # using the compiler binary.
+  outputs = [ "out" "dev" ];
+
+  src = fetchFromGitHub {
+    owner = "Microsoft";
+    repo = "DirectXShaderCompiler";
+    rev = "b640fa4ba4afb8eb9f15b4a34ab5187c7d94de97";
+    sha256 = "0ccfy1bfp0cm0pq63ri4yl1sr3fdn1a526bsnakg4bl6z4fwrnnj";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ cmake git python3 ];
+
+  configurePhase = ''
+    # Requires some additional flags to cmake from a file in the repo
+    additionalCMakeFlags=$(< utils/cmake-predefined-config-params)
```

Comment:
There is no way to use a variable from nix instead of this?

Reply:
Don't think so, unless one uses IFD or just replicates the contents of that file.

Reply:
(or builtins.readFile I suppose, but that would still require fetching the source at instantiation time.